fdmi/plugins/m0scheduler

```diff
@@ -17,128 +17,277 @@
+    kv = {
+        'fid': fid,
+        'cr_key': cr_key,
+        'cr_val': cr_val
+    }
+    # kv = {'fid': kvs["fid"], 'cr_key': kvs["cr_key"], 'cr_val': kvs["cr_val"]}
+    flag = True
+
+    # check dedup records
+    for kv_i in kv_records:
+        if kv['cr_key'] == kv_i['cr_key'] and kv['cr_val'] != "0":
+            # print(kv_records)
+            print('dedup: {} {}'.format(kv_i, kv), file=sys.stderr)
+            flag = False
+            break
+
+    # if record is not dedupable, add to cache
+    if flag and kv['cr_val'] != "0":
+        print('new kv: {}'.format(kv), file=sys.stderr)
+        kv_records.append(kv)
+
+
+# handle SIGNT and SIGTERM
 def signal_handler(sig, frame):
-	print("CTRL + C pressed")
-	process.send_signal(sig)
-	process.wait()
-	#print(kv_records)
-	sys.exit(0)
+    print('CTRL + C pressed', file=sys.stderr)
+    print('Stopping all m0sched processes...')
+    for p in processes:
+        p.send_signal(sig)
+        p.wait()
+
+    # print(kv_records)
+    sys.exit(0)
+
 
 def execute_command_and_get_output(cmd):
     proc = subprocess.Popen(cmd, stdin=subprocess.PIPE,
-                                    stdout=subprocess.PIPE,
-                                    stderr=subprocess.PIPE,
-                                    shell=True)
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True)
     stdout, stderr = proc.communicate()
     output_var = str(stdout, 'utf-8')
     return output_var
 
 
-#m0sched -l 192.168.52.53@tcp:12345:4:1 -h 192.168.52.53@tcp:12345:1:1 -p 0x7000000000000001:0x37 -f 0x7200000000000001:0x19
-def get_m0sched_cmd():
-	#m0sched = "/home/931442/cortx-motr/fdmi/plugins/m0sched"
-	m0sched = "m0sched"
-	opts = " -l " + local_endpoint + " -h " + ha_endpoint + " -p " + profile_fid + " -f " + process_fid + " -g " + fdmi_filter_fid
-	return m0sched + opts
-
-#get local ep, ha ep, profile fid and process fid
-def get_cluster_info():
-    ret = 0
-    global profile_fid
-    global ha_endpoint
-    global local_endpoint
-    global process_fid
-    global fdmi_filter_fid
-    cmd = 'hctl status --json'
-    cluster_info = execute_command_and_get_output(cmd)
-    cluster_js = json.loads(cluster_info)
-    if (cluster_info != None):
-        profile_fid = str(cluster_js["profiles"][0]["fid"])
-        nodes_data = cluster_js["nodes"]
-        for svcs in nodes_data[0]['svcs']:
-            if svcs['name'] == 'hax':
-                ha_endpoint = str(svcs['ep'])
-
-            #m0sched alway uses first m0client
-            if svcs['name'] == 'm0_client':
-                local_endpoint = str(svcs['ep'])
-                process_fid = str(svcs['fid'])
-                break
+# m0sched -l 192.168.52.53@tcp:12345:4:1 -h 192.168.52.53@tcp:12345:1:1 -p 0x7000000000000001:0x37
+# -f 0x7200000000000001:0x19
+def get_scheduler_cmd(args):
+    # m0sched = "/home/931442/cortx-motr/fdmi/plugins/m0sched"
```

Comment:
remove commented code, here and other places.

Reply:
Maybe we can:
 # m0sched = "/path/to/cortx-motr/fdmi/plugins/m0sched"

Reply:
agreed